src/components/accessories/dateField/DateField.tsx

```diff
@@ -1,53 +1,49 @@
-import React, { FunctionComponent, memo, useEffect, useState } from "react";
-import { TextField } from "@material-ui/core";
-import { useRifm } from "rifm";
-import { formatToDate } from "./utils";
+import React, { FunctionComponent, useEffect, useState } from "react";
+import { KeyboardDatePicker as DatePicker, MuiPickersUtilsProvider as DatePickerWrapper } from '@material-ui/pickers';
+import DateFnsUtils from '@date-io/date-fns';
 import { IProps } from "./types";
 import "./styles.scss";
 
 const DateField: FunctionComponent<IProps> = ({
   fieldName,
   fieldValue,
-  label,
-  isValid,
-  errorText,
-  onBlur,
+  disableFuture,
   disabled,
-  InputProps,
+  label,
+  theme,
+  format,
+  onChange
 }) => {
-  const [value, setValue] = useState("");
-  const rifm = useRifm({
-    value,
-    onChange: setValue,
-    format: formatToDate,
-  });
+
+  const [value, setValue] = useState<Date | null>(new Date(+fieldValue));
 
   useEffect(() => {
-    setValue(fieldValue);
-  }, [fieldValue]);
+    setValue(new Date(+fieldValue));
+  }, [fieldValue])
 
-  const handleOnBlur = (e: any) => {
-    onBlur(e, value);
+  const handleDateChange = (date: Date | null) => {
+    onChange(date);
```

Comment:
It should update the Formik values using onBlur in order to avoid unnecessary re-renders.

Reply:
Unfortunately I cannot fire the onBlur event through the Calendar Widget. To avoid this I use the onChange function that calls the dateFieldHandleOnChange callback in order to proper update Formik.

Reply:
🤔 I see. It's something that regards performance, so I'd say we must figure something out further.